src/api/v4/layer/layer.js

```diff
@@ -20,65 +22,122 @@ var CartoDBLayer = require('../../../geo/map/cartodb-layer');
  * @api
  * @memberof carto.layer
  */
-function Layer (id, source, style) {
-  if (typeof style === 'undefined') {
+function Layer (id, source, style, options) {
+  if (typeof options === 'undefined') {
```

Comment:
still dont like this

Reply:
The examples provided in the doc doesnt work:


```javascript

// @example 1 Wont work, opts is undefined
new carto.layer.Layer('cities', citiesSource, citiesStyle);

// @example 2 wont work, style will point to the dataset
 new carto.layer.Layer(citiesSource, citiesStyle);

```

Reply:
Yes, it doesn't work :(